go-controller/pkg/ovn/ovn.go

```diff
@@ -555,6 +561,56 @@ func (oc *Controller) WatchNetworkPolicy() error {
 	return err
 }
 
+// WatchEgressFirewall starts the watching of egressfirewall resource and calls
+// back the appropriate handler logic
+func (oc *Controller) WatchEgressFirewall() error {
+	_, err := oc.watchFactory.AddEgressFirewallHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			egressFirewall := obj.(*egressfirewall.EgressFirewall)
+			errList := oc.addEgressFirewall(egressFirewall)
+			for _, err := range errList {
+				klog.Error(err)
+				egressFirewall.Status.EgressFirewallObjectStatus = "EgressFirewall not applied correctly, please check logs"
```

Comment:
This looks awkward, could you just set the value once?

Reply:
done